### PR TITLE
EI-780 display estimated payment month/year

### PR DESCRIPTION
**Pull request by ruslanss**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/DataAccess.cs [hunk 1]
```diff
@@ -52,25 +53,10 @@ public async Task SetupAccount(Account account)
...
-                    application.Id,
-                    application.AccountId,
-                    application.AccountLegalEntityId,
-                    application.DateCreated,
-                    Status = application.Status.ToString(),
-                    application.DateSubmitted,
-                    application.SubmittedByEmail,
-                    application.SubmittedByName
-                });
+            return await dbConnection.InsertWithEnumAsString(entity);
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/DataAccess.cs:59]: Dapper has custom TypeMappers e.g.

```
   public class EarningsTypeHandler : SqlMapper.TypeHandler<EarningType>
    {        
        public override EarningType Parse(object value)
        {
            Enum.TryParse(value.ToString(), out EarningType earningType);

            return earningType;
        }

        public override void SetValue(IDbDataParameter parameter, EarningType value)
        {
            parameter.Value = value.ToString();
        }
    }
```

    

   and register with
    
    SqlMapper.AddTypeHandler(typeof(EarningType), new EarningsTypeHandler());
    
    Note doesn't work with Dapper Contrib but will work with SqlMapper.

**Review by Stephek — CHANGES_REQUESTED** [review]

**Review by Stephek — APPROVED** [review]

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Steps/ApplicationsForAccountRequestedSteps.cs [hunk 2]
```diff
@@ -61,12 +58,13 @@ public void ThenTheApprenticeshipApplicationsAreReturned()
             var apprenticeshipApplication = _apiResponse.ApprenticeApplications.First();
 
             apprenticeshipApplication.AccountId.Should().Be(_account.Id);
-            apprenticeshipApplication.ApplicationId.Should().Be(_application.Id);
-            apprenticeshipApplication.FirstName.Should().Be(_apprenticeship.FirstName);
-            apprenticeshipApplication.LastName.Should().Be(_apprenticeship.LastName);
+            //apprenticeshipApplication.ApplicationId.Should().Be(_application.Id); //TODO: This needs resolving as the bank details needs the original application id
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Steps/ApplicationsForAccountRequestedSteps.cs:61]: TODO :

**Review by Stephek — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/DataAccess.cs (hunk 1 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/DataAccess.cs:59]: i've added a generic method which reads the the property attributes to work out whether it needs to insert enum as string or not. please take a look.

Comment (Stephek) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/DataAccess.cs:59]: Nice one.

Can you change the new method to match the InsertAsync  e.g.

```
public static Task<int> InsertAsync<T>(this IDbConnection connection, T entityToInsert, bool enumToString = false, IDbTransaction transaction = null, int? commandTimeout = null, ISqlAdapter sqlAdapter = null) where T : class
        {
            if (enumToString)
            {
                ...
            }
            else
            {
                return Dapper.Contrib.Extensions.SqlMapperExtensions.InsertAsync(connection, entityToInsert, transaction, commandTimeout, sqlAdapter);
            }
        }

```

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/DataAccess.cs]: good idea!

**Pull request closed without merging**